csrc/executor.cpp

```diff
@@ -257,11 +260,30 @@ void FusionExecutor::compileFusion(
   NVF_ERROR(
       !fusion->outputs().empty(), "No output found for this kernel, aborting.");
 
+  // TODO: refactor the options_ passed through
+  options_.device = c10::Device(c10::DeviceType::CUDA, args.getDeviceIndex());
```

Comment:
Good question. I moved this code as is from further down without thinking too much about it. 

`communicator` is optional. It's very likely null for single-GPU execution. So `args.getDeviceIndex()` is more available. 

For the future, I think they (using `communicator->deviceId()` and using `args.getDeviceIndex()`) are both wrong because compilation can be done on a different machine than the machine that executes the fusion.

Reply:
> because compilation can be done on a different machine than the machine that executes the fusion.

Do we really support that?

Reply:
No, we don't. We may need this in the future but it's too speculative at this moment to worry about.